src/JWT/TokenExpiredException.cs

```diff
@@ -1,12 +1,38 @@
 ﻿using System;
+using System.Collections.Generic;
 
 namespace JWT
 {
     public class TokenExpiredException : Exception
     {
-        public TokenExpiredException(string message)
+       private const string PayloadDataKey = "PayloadData";
+       private const string ExpirationKey = "Expiration";
+
+       public TokenExpiredException(string message)
             : base(message)
         {
         }
+
+        public Dictionary<string, object> PayloadData
+        {
+            get
+            {
+               if (Data.Contains(PayloadDataKey))
```

Comment:
Can you move this snippet to a helper method? To reuse, also make both getters one-liners. Also do we really need to check a key to exist? I'd let it throw.